zerver/actions/message_flags.py

```diff
@@ -264,6 +265,7 @@ def do_update_message_flags(
     is_adding = operation == "add"
     flagattr = getattr(UserMessage.flags, flag)
     flag_target = flagattr if is_adding else 0
+    skipped_marking_unread_stream_ids = None
 
     with transaction.atomic(savepoint=False):
         if flag == "read" and not is_adding:
```

Comment:
I'm not entirely sure about how to structure the changes in this commit. I was also considering separating this if block into a separate function that is called in `views > message_flags.py > update_message_flags_for_narrow`, like this:

````
if flag == "read" and operation == "remove":
    (messages, skipped_marking_unread_stream_ids) = filter_messages_for_marking_as_unread(user_profile, messages)
   ````
Returning `skipped_marking_unread_stream_ids` from `do_update_message_flags` might seem unusual. If we choose to do this, we will also need to call the function in `views > message_flags.py > update_message_flags`.

Reply:
I think these changes are fine for now and we can update later if  Tim suggests to do something else now.